src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,6 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public abstract boolean isMutating();
```

Comment:
Better you add a header comment to both this method and overriding methods since this method returns a very sensitive information and method name does not give the complete picture of the responsibility.